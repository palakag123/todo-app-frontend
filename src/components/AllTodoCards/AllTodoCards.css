.all-cards-outer-container{
    max-height: 500px;
    min-height: 500px;
    max-width: 90%;
    min-width: 90%;
    border: 4px solid white;
    display: flex;
    justify-content: center;
}
.all-cards-container{
    height: 98%;
    width: 98%;
    overflow-y: scroll;
    max-height: 500px;
}

/* width */
.all-cards-container::-webkit-scrollbar {
    width: 20px;
  }
  
  /* Track */
  .all-cards-container::-webkit-scrollbar-track {
    background: rgb(222,248,255); 
    border-radius: 10px;
    display: none;
  }
   
  /* Handle */
  .all-cards-container::-webkit-scrollbar-thumb {
    background: white; 
    border-radius: 10px;
  }