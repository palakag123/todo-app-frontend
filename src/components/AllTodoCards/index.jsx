import React from "react";
import PropTypes from "prop-types";
import "./AllTodoCards.css";
import TodoCard from "../TodoCard";

function AllTodoCards({ todos }) {
  return (
    <div className="all-cards-outer-container" data-testid="allCards">
      <div className="all-cards-container" data-testid="allCardsInner">
        {todos?.map((todo) => (
          <TodoCard todo={todo} key={todo?.id} data-testid="todoCard" />
        ))}
      </div>
    </div>
  );
}
AllTodoCards.propTypes = {
  todos: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number.isRequired,
      todo: PropTypes.string.isRequired,
      // eslint-disable-next-line comma-dangle
    })
  ).isRequired,
};
export default AllTodoCards;
