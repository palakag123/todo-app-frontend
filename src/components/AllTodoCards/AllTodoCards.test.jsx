import React from "react";
import { render, screen } from "@testing-library/react";
import AllTodoCards from "./index";

const mockNavigate = jest.fn();
jest.mock("react-router-dom", () => ({ useNavigate: () => mockNavigate }));

describe("AllTodoCards", () => {
  const mockTodos = [
    { id: 1, todo: "first todo list item" },
    { id: 2, todo: "second todo list item" },
  ];
  beforeEach(() => {
    render(<AllTodoCards todos={mockTodos} />);
  });
  it("should take a snapshot for AllTodoCards", () => {
    const { asFragment } = render(<AllTodoCards todos={mockTodos} />);
    expect(asFragment(<AllTodoCards todos={mockTodos} />)).toMatchSnapshot();
  });
  it("should have the required elements when AllTodoCards is rendered", () => {
    expect(screen.findAllByText("first todo list item")).toBeTruthy();
    expect(screen.findAllByText("second todo list item")).toBeTruthy();
    expect(screen.getByTestId("allCards")).toBeTruthy();
    expect(screen.getByTestId("allCardsInner")).toBeTruthy();
  });
});
