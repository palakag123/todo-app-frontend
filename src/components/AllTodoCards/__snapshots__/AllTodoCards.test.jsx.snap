// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllTodoCards should take a snapshot for AllTodoCards 1`] = `
<DocumentFragment>
  <div
    class="all-cards-outer-container"
    data-testid="allCards"
  >
    <div
      class="all-cards-container"
      data-testid="allCardsInner"
    >
      <div
        class="todo-card"
        data-testid="todoCard"
      >
        <p
          id="todo-title"
        >
          first todo list item
        </p>
      </div>
      <div
        class="todo-card"
        data-testid="todoCard"
      >
        <p
          id="todo-title"
        >
          second todo list item
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
